core/typedefs.h

```diff
@@ -154,7 +154,23 @@ inline bool is_power_of_2(const T x) {
 }
 
 // Function to find the next power of 2 to an integer.
-static _FORCE_INLINE_ unsigned int next_power_of_2(unsigned int x) {
+static constexpr uint64_t next_power_of_2(uint64_t x) {
```

Comment:
All these should be `constexpr` without `static`. Sorry, I think I may have mis-suggested that in my earlier comment 😅

`static` for functions is a keyword to the compiler to guarantee each translation unit gets their own version of this function. It's not what we want here (honestly, it's rarely what anyone wants...).

```suggestion
constexpr uint64_t next_power_of_2(uint64_t x) {
```